timeseries-aggregator/src/test/scala/com/expedia/www/haystack/metricpoints/feature/tests/aggregation/metrics/MetricSpec.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ *
+ *     Copyright 2017 Expedia, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ *
+ */
+
+package com.expedia.www.haystack.metricpoints.feature.tests.aggregation.metrics
+
+import com.expedia.www.haystack.metricpoints.aggregation.metrics.{CountMetric, HistogramMetric, Metric, MetricFactory}
+import com.expedia.www.haystack.metricpoints.aggregation.rules.MetricRuleEngine
+import com.expedia.www.haystack.metricpoints.entities.Interval.Interval
+import com.expedia.www.haystack.metricpoints.entities.MetricType.MetricType
+import com.expedia.www.haystack.metricpoints.entities.{HistogramType, Interval, MetricPoint, MetricType}
+import com.expedia.www.haystack.metricpoints.feature.FeatureSpec
+import org.HdrHistogram.IntHistogram
+
+class MetricSpec extends FeatureSpec with MetricRuleEngine {
+
+  val DURATION_METRIC_NAME = "duration"
+  val TOTAL_METRIC_NAME = "total-spans"
+  val INVALID_METRIC_NAME = "invalid_metric"
+  val SERVICE_NAME = "dummy_service"
+  val OPERATION_NAME = "dummy_operation"
+
+  object TagKeys {
+    val OPERATION_NAME_KEY = "operationName"
+    val SERVICE_NAME_KEY = "serviceName"
+  }
+
+  val keys = Map(TagKeys.OPERATION_NAME_KEY -> OPERATION_NAME,
+    TagKeys.SERVICE_NAME_KEY -> SERVICE_NAME)
+
+  feature("Creating a Metric (Count / Histogram") {
```

Comment:
is it a common feature test for metric factory and metric rule engine? I think  metric factory and metric rule engine should be treated as different features. the metric rule engine feature tests should just talk about fetching the right metric type, and the metric factory should talk about fetching the right metric given a metric type, so we can break this test class into two classes